lib/helpers/stake/gateway_composer/StakeHelper.js

```diff
@@ -122,17 +130,25 @@ class StakeHelper {
       throw Error('Value token abi is not provided');
     }
 
-    const txOptions = {
+    const defaultOptions = {
       from: facilitator,
       to: oThis.gatewayComposer,
-      gas: '500000',
     };
 
-    const contract = new web3.eth.Contract(valueTokenAbi, valueTokenContractAddress, txOptions);
+    if (txOptions) {
```

Comment:
Should we add validations for from address in  txOptions ?

Reply:
Same as the other comment (think this is out of scope of the PR).